src/Beacon.jsx

```diff
@@ -177,8 +204,15 @@ export class Beacon extends Component {
     return targetClone;
   }
 
-  renderBeacon(persistent) {
-    if (this.state.inactive || (persistent && !this.state.hash)) {
+  getHash() {
+    const { tooltipText, id } = this.props;
+    const text = typeof(tooltipText) === 'string' ? tooltipText : ReactDOMServer.renderToStaticMarkup(tooltipText);
```

Comment:
Yeah, no worries I guess it would be premature optimization anyway... I am nitpicking sometimes :P